uk.co.compucorp.civicrm.hrleaveandabsences/tests/phpunit/CRM/HRLeaveAndAbsences/BAO/ContactWorkPatternTest.php

```diff
@@ -463,4 +467,141 @@ public function getAllForPeriodReturnsOnlyTheContactWorkPatternsOverlappingTheGi
     $this->assertInstanceOf(ContactWorkPattern::class, $contactWorkPatterns[1]);
     $this->assertEquals($contactWorkPattern3->id, $contactWorkPatterns[1]);
   }
+
+  public function testGetContactsForPeriodReturnsOnlyTheContactsWithWorkPatternsOverlappingTheGivenPeriod() {
+    $contactID1 = 1;
+    $contactID2 = 2;
+    $contactID3 = 3;
+    $workPatternID = 1;
+
+    $contactWorkPattern1 = ContactWorkPatternFabricator::fabricate([
+      'contact_id' => $contactID1,
+      'pattern_id' => $workPatternID,
+      'effective_date' => CRM_Utils_Date::processDate('2015-01-10'),
+    ]);
+
+    $contactWorkPattern2 = ContactWorkPatternFabricator::fabricate([
+      'contact_id' => $contactID2,
+      'pattern_id' => $workPatternID,
+      'effective_date' => CRM_Utils_Date::processDate('2015-01-15'),
+    ]);
+
+    $contactWorkPattern3 = ContactWorkPatternFabricator::fabricate([
+      'contact_id' => $contactID1,
+      'pattern_id' => $workPatternID,
+      'effective_date' => CRM_Utils_Date::processDate('2015-01-20'),
+    ]);
+
+    ContactWorkPatternFabricator::fabricate([
+      'contact_id' => $contactID3,
+      'pattern_id' => $workPatternID,
+      'effective_date' => CRM_Utils_Date::processDate('2015-01-30'),
+    ]);
+
+    $contacts = ContactWorkPattern::getContactsForPeriod(
+      new DateTime('2015-01-14'),
+      new DateTime('2015-01-29')
+    );
+
+    // ContactWork Patterns 1,2 and 3  overlapps the given date range
+    // Unique Contacts attached to these Contact work patterns will be returned
+    $this->assertCount(2, $contacts);
+    sort($contacts);
+    $this->assertEquals($contacts, [$contactID1, $contactID2]);
+  }
+
+  public function testGetContactsForPeriodReturnsCorrectlyWhenWorkPatternsOverlappsTheGivenPeriodAndForAGivenWorkPatternID() {
```

Comment:
ReturnsCorrectly is too vague. Can't we have something similar to the previous method? Something like "...ReturnsOnlyTheContactsWith..."